File: 206. Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */


ListNode* reverseLinkList(ListNode* node) { 
    
    ListNode* head = NULL;
    while( node ) { 
        ListNode* currentNode = new ListNode(node->val);
        if( head == NULL ) head = currentNode;
        else { 
            currentNode->next = head;
            head = currentNode;
        }
        node = node->next;
    }
    return head;
}
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        return reverseLinkList(head);
    }
};

File: python/Lakshay/206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        root = None
        prv = None
        while head:
            curNode = ListNode(head.val)
            root = curNode
            root.next = prv
            prv = curNode
            head = head.next
        
        return root
